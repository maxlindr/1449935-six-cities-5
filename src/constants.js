export const OfferCardType = {
  FAVORITE: `favorite`,
  NEAR_PLACE: `near_place`,
  CITIES: `cities`
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
  PENDING: `PENDING`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVORITES: `/favorites`,
  ERROR: `/error`,
  OFFER: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};

export const ErrorMessage = {
  NOT_FOUND: `Resource not found`,
  GENERAL: `Oops! Something went wrong`,
  UNAUTHORIZED: `Unauthorized`,
};
