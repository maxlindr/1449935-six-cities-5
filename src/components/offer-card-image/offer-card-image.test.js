import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OfferCardImage from './offer-card-image';
import {OfferCardType} from '../../constants';
import offer from '../../../__mocks__/offer-mock';

const renderer = new ShallowRenderer();

test(`OfferCardImage должен корректно отображаться`, () => {
  renderer.render(
      <OfferCardImage
        offer={offer}
        cardType={OfferCardType.CITIES}
      />
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
