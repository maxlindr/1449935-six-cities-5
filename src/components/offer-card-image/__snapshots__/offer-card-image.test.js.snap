// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCardImage должен корректно отображаться 1`] = `
<React.Fragment>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <Link
      to="/offer/1"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
        width={260}
      />
    </Link>
  </div>
</React.Fragment>
`;
